### ISSUE TITLE
Address "Entity too large" error 

### ISSUE THREAD

**Issue by cecilia-donnelly** [OpenTechStrategies/psm#263]
When entering an enrollment as an Audiologist provider on the sandbox, I entered a large .xcf file as the License.  It was 2.8 MB.  I wasn't able to trigger the error locally, and it appears to be an nginx problem:

![screenshot from 2017-07-11 12-07-38](https://user-images.githubusercontent.com/1497818/28080501-d709bd9a-6631-11e7-97bc-1f5e5fa899f5.png)

I'm not sure how big the uploaded file needs to be to get this error.  We should set a sensible limit on file upload size, and give a more attractive error to users if they exceed it.  The error should show up on the same page, and allow them to choose a different file.

**Comment by jasonaowen** [comment]
You are right that this is an nginx error. [The nginx default maximum upload size is 1 MB](https://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size). Uploads larger than that trigger the error you saw.

[Our application also has a maximum upload size](thttps://github.com/OpenTechStrategies/psm/blob/0987da3c32b7fca5dd606290d07464197c0072b2/psm-app/services/src/main/resources/cms.properties#L219); currently [it is set to about 488 MB](https://github.com/OpenTechStrategies/psm/blob/0987da3c32b7fca5dd606290d07464197c0072b2/psm-app/services/src/main/resources/cms.properties#L219). If you upload something larger, you get an even worse error message:

![Stack trace](https://user-images.githubusercontent.com/1494855/28084757-a93c6b54-6647-11e7-88f7-52b5174dbf59.png)

1 MB is clearly too small; 488MB may be too large. When acting as a reverse proxy, as nginx does in our sandbox setup, nginx will temporarily store uploads on disk; in our environment, in the directory `/var/lib/nginx/tmp/client_body`. If we choose to recommend nginx for production, this will have an impact on the storage requirements of our application, in addition to the database size. Finding a reasonable upper limit is hard!

**Comment by jasonaowen** [comment]
Currently, we are using a simple HTML form with a file `<input>`. The server discovers any errors in the form input after the browser `POST`s the form; layers include nginx, Spring, and drools validation. If there are any validation errors, they are shown on the same page.

There are a few ways to show friendly errors if the uploaded file is too large:
- nginx can be configured with a static error page. This is simple, and allows for a slightly better user experience, but the flow will be interrupted.
- [Spring can let you handle the error](https://stackoverflow.com/a/3230352/25637) by implementing [HandlerExceptionResolver](http://docs.spring.io/spring/docs/3.0.5.RELEASE/javadoc-api/org/springframework/web/servlet/HandlerExceptionResolver.html). After briefly trying to implement this, it appears that the way the application is written will make that difficult, as we seem not to have the information we need to reconstruct the page available at error handling time. We can make a simple error page (more dynamic than in nginx), but it's not obvious to me now how to show the previous page with an error message, the way the validation errors are displayed.
- We can rework the attachment workflow to do an AJAX upload to an API endpoint, and handle the response in JavaScript.

These are in ascending order of work required.

**Comment by jasonaowen** [comment]
For now, I'll remove the nginx limit in the sandbox. We can work on this more when this is the top priority.

**Comment by jvasile** [comment]
I made some notes about this in the INSTALL.md.  Anybody deploying should be aware of it.

I think the TODO here is to set a sane default in cms.properties.  It should be quite large.  Multipage PDF scans can get hefty and we can't really expect providers to slim down their docs.  That would take knowledge and tooling we can't expect people to have.  A file-to-large error encountered in normal use is a bug IMO.  If we are really concerned about this, we can accept compressed files and automatically unzip them for providers.

**Comment by jasonaowen** [comment]
It is in bytes; I assume they ran into a similar problem and raised the size without fixing the comment.

**Comment by jasonaowen** [comment]
@jvasile: fixed in 0b70deeeee9bfef718b0703ae250a6eb06a32790.
